src/Corax/Queries/SuggestionTermProvider.cs

```diff
@@ -1,140 +1,481 @@
+                        break; // No enough space to put it here. 
+
+                    float score = nGramDistance.GetDistance(term, termKey);
+                    if (score < provider._distance)
+                        continue;
+
+                    termKey.CopyTo(auxTerms);
+                    tokens[tokenTaken].Offset = totalTermsBytes;
+                    tokens[tokenTaken].Length = (uint)termKey.Length;
+
+                    scores[tokenTaken] = score;
+
+                    auxTerms = auxTerms.Slice(termKey.Length);
+                    totalTermsBytes += termKey.Length;
+
+                    tokenTaken++;
+                    if (tokenTaken >= take)
+                        break;
+                }
+
+                terms = terms.Slice(0, totalTermsBytes);
+                tokens = tokens.Slice(0, tokenTaken);
+                scores = scores.Slice(0, tokenTaken);
+
+                return;
+
+
+            NoResults:
+                terms = terms.Slice(0, 0);
+                tokens = tokens.Slice(0, 0);
+                scores = scores.Slice(0, 0);
+            }
+
+            static void Next(ref SuggestionTermProvider<TDistanceProvider> provider, ref Span<byte> terms, ref Span<Token> tokens, ref Span<float> scores)
+            {
+                int fieldId = provider._fieldId;
+                var allocator = provider._searcher.Allocator;
+                var term = provider._term;
+                var maxDistance = provider._distance;
+
+                // We get the actual field tree. 
+                var fields = provider._searcher.Transaction.CreateTree(FieldsSlice);
+                var fieldTree = fields.CompactTreeFor(provider._binding.FieldName);
+                if (fieldTree == null)
+                    goto NoResults;
+
+                // For each term to look for, we will look at every potential hit using fuzzy searching with low threshold unless we are doing it
+                // using Levenshtein, in which case we just do it properly.                
+                var currentTermSlice = provider._term;                                
+                var iter = fieldTree.FuzzyIterate(term, typeof(TDistanceProvider) == typeof(LevenshteinDistance) ? provider._distance : 0.3f);
+                iter.Seek(currentTermSlice);
+
+                var values = new FastList<(Slice Term, float)>();
+
+                TDistanceProvider distance = provider._distanceProvider;                
+                while (iter.MoveNext(out Slice key, out var _, out float score))
+                {
+                    // The original distance is Levenshtein, therefore we dont need to recompute it. 
+                    if (typeof(TDistanceProvider) != typeof(LevenshteinDistance))
+                    {
+                        score = distance.GetDistance(term, key);
+                        if (score >= provider._distance)
+                            continue;
```

Comment:
shouldn't the if here be outside? can we have a `score` that is greater than `_distance` ?